src/object.rs

```diff
@@ -67,9 +77,15 @@ pub struct ObjectDescription {
     pub passable: bool,
 
     /// Arbitrary data associated with this object
-    pub associated_data: Vec<u8>,
+    #[serde(bound(deserialize = "T: AssociatedObjectData"))]
+    pub associated_data: T,
 }
 
+/// Required bounds for [`ObjectDescription::associated_data`].
+/// This trait is automatically implemented for all types that satisfy the supertraits.
+pub trait AssociatedObjectData =
+    Clone + Debug + PartialEq + Serialize + for<'de> Deserialize<'de> + 'static;
```

Comment:
```suggestion
    Clone + Debug + PartialEq + Serialize + for<'a> Deserialize<'a> + 'static;
```

Reply:
The lifetime on the deserialize trait is called `'de` thats why I called it that too.